.editorconfig

```
@@ -121,15 +153,25 @@ csharp_space_between_method_declaration_parameter_list_parentheses = false
 csharp_space_between_parentheses = false
 csharp_space_between_square_brackets = false
 
+# Analyzers
+dotnet_code_quality.ca1802.api_surface = private, internal
+dotnet_code_quality.ca2208.api_surface = public
+
+# License header
+file_header_template = Copyright (c) .NET Foundation and contributors. All rights reserved.\nLicensed under the MIT license. See LICENSE file in the project root for full license information.\n
+
 # C++ Files
 [*.{cpp,h,in}]
 curly_bracket_next_line = true
 indent_brace_style = Allman
 
 # Xml project files
-[*.{csproj,vcxproj,vcxproj.filters,proj,nativeproj,locproj}]
+[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,nativeproj,locproj}]
 indent_size = 2
 
+[*.{csproj,vbproj,proj,nativeproj,locproj}]
+charset = utf-8
```

I copied the editorconfig exactly the same from dotnet/runtime.
Should I use utf-8-bom here instead? and probably give dotnet/runtime a PR too?

cc: @sharwell

---

Yes, it should be utf-8-bom. I can't say whether or not to sent a PR to dotnet/runtime.

```suggestion
charset = utf-8-bom
```

---

@sharwell Will this require a bulk PR like the one you did in dotnet/docs a short time ago?

---

I also just noticed it's not set for .cs and .vb files. The charset should include these extensions right?

---

It may or may not. The project system ignores this setting and always saves with BOM, so the only way it would be wrong is if someone edited it in another text editor.

---

I actually often use notepad++ for very small changes as it's very light and fast :smile:
I'd leave this up to @wli3 to decide.

---

let's keep -bom for now